website/addons/gitlab/tests/test_views.py

```diff
@@ -0,0 +1,560 @@
+        self.node_settings.registration_data = {
+            'branches': [
+                branch.to_json()
+                for branch in gitlab_mock.branches.return_value
+            ]
+        }
+        self.node_settings.owner.is_registration = True
+        with assert_raises(HTTPError):
+            utils.get_refs(self.node_settings, branch='nothere')
+
+    # Tests for _check_permissions
+    # make a user with no authorization; make sure check_permissions returns false
+    def test_permissions_no_auth(self):
+        gitlab_mock = self.gitlab
+        # project is set to private right now
+        connection = gitlab_mock
+        non_authenticated_user = UserFactory()
+        non_authenticated_auth = Auth(user=non_authenticated_user)
+        branch = 'master'
+        assert_false(check_permissions(self.node_settings, non_authenticated_auth, connection, branch))
+
+    # make a repository that doesn't allow push access for this user;
+    # make sure check_permissions returns false
+    @mock.patch('website.addons.gitlab.model.GitLabUserSettings.has_auth')
+    @mock.patch('website.addons.gitlab.api.GitLabClient.repo')
+    def test_permissions_no_access(self, mock_repo, mock_has_auth):
+        gitlab_mock = self.gitlab
+        mock_has_auth.return_value = True
+        connection = gitlab_mock
+        branch = 'master'
+        mock_repository = mock.NonCallableMock()
+        mock_repository.user = 'fred'
+        mock_repository.repo = 'mock-repo'
+        mock_repository.to_json.return_value = {
+            'user': 'fred',
+            'repo': 'mock-repo',
+            'permissions': {
+                'push': False,  # this is key
+            },
+        }
+        mock_repo.return_value = mock_repository
+        assert_false(check_permissions(self.node_settings, self.consolidated_auth, connection, branch, repo=mock_repository))
+
+    # make a branch with a different commit than the commit being passed into check_permissions
+    @mock.patch('website.addons.gitlab.model.GitLabUserSettings.has_auth')
+    def test_permissions_not_head(self, mock_has_auth):
+        gitlab_mock = self.gitlab
+        mock_has_auth.return_value = True
+        connection = gitlab_mock
+        mock_branch = mock.NonCallableMock()
+        mock_branch.commit.sha = '67890'
+        sha = '12345'
+        assert_false(check_permissions(self.node_settings, self.consolidated_auth, connection, mock_branch, sha=sha))
+
+    # make sure permissions are not granted for editing a registration
+    @mock.patch('website.addons.gitlab.model.GitLabUserSettings.has_auth')
+    def test_permissions(self, mock_has_auth):
+        gitlab_mock = self.gitlab
+        mock_has_auth.return_value = True
+        connection = gitlab_mock
+        self.node_settings.owner.is_registration = True
+        assert_false(check_permissions(self.node_settings, self.consolidated_auth, connection, 'master'))
+
+    def check_hook_urls(self, urls, node, path, sha):
```

Comment:
it is used by tests on https://github.com/CenterForOpenScience/osf.io/pull/5929/files/1fbc939831c5994ea0088cd0303b11ce45868563#diff-7bf832417d62003700f5f44838fc208aR289 and https://github.com/CenterForOpenScience/osf.io/pull/5929/files/1fbc939831c5994ea0088cd0303b11ce45868563#diff-7bf832417d62003700f5f44838fc208aR316